pennylane/resources/factorization.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the functions needed for two-electron tensor factorization.
+"""
+from pennylane import numpy as np
+
+
+def factorize(two, tol):
```

Comment:
I was thinking that we don't offer the possibility of doing just a single factorization. Would it make sense to add an additional tolerance for the eigenvector factorization that takes a default value, but could also be passed independently by the user?

Reply:
Two distinct thresholds added.